### DISCUSSION TITLE
I downloaded MAS on my Mac from the website and I keep getting this...

### DISCUSSION THREAD

**Discussion by N-1C0lE** [Monika-After-Story/MonikaModDev#9862] [QA - tech support + general]
**An exception has occurred**

I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/definitions.rpy", line 9, in script
  File "game/definitions.rpy", line 10, in <module>
ValueError: list.remove(x): x not in list

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/definitions.rpy", line 9, in script
  File "/Users/NopeSorry/Desktop/DDLC.app/Contents/Resources/autorun/renpy/ast.py", line 814, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "/Users/NopeSorry/Desktop/DDLC.app/Contents/Resources/autorun/renpy/python.py", line 1719, in py_exec_bytecode
    exec bytecode in globals, locals
  File "game/definitions.rpy", line 10, in <module>
  File "/Users/NopeSorry/Desktop/DDLC.app/Contents/Resources/autorun/renpy/python.py", line 618, in do_mutation
    return method(self, *args, **kwargs)
ValueError: list.remove(x): x not in list

Darwin-21.4.0-x86_64-i386-64bit
Ren'Py 6.99.12.4.2187
Monika After Story 0.12.13

**Comment by N-1C0lE** [comment]
If anyone knows how to fix please tell me, I'm new to this and it's pretty confusing :C

**Comment by Booplicate** [comment]
That's not in our code, so perhaps you installed something else, or already had another mod in your install.

**Comment by Booplicate** [comment]
Are you using RenPy SDK?

**Reply by N-1C0lE** [reply]
I don't really know what RenPy SDK is as I said I was new to this lol. Is it necessary to play MAS and should I install it?

**Reply by Booplicate** [reply]
No, you shouldn't use SDK. Try to make a new clear install of DDLC and install MAS into it.

**Reply by N-1C0lE** [reply]
I tried installing and reinstalling it multiple times though

**Reply by Booplicate** [reply]
Show where you have DDLC installed and show where you extracted the mod files.

**Reply by N-1C0lE** [reply]
sorry I've been on break for a while and couldn't get to you, I wanted to ask you what you meant by "where you have DDLC installed and show where you extracted the mod files"

**Reply by Booplicate** [reply]
Just try to reinstall. [Make a new DDLC install](https://ddlc.moe/), then [get MAS installer and install MAS](https://monikaafterstory.com/releases.html).

**Reply by N-1C0lE** [reply]
Same thing happened

**Reply by Booplicate** [reply]
Did it warn you that you selected the wrong directory during installation?

**Reply by N-1C0lE** [reply]
there was no warning, every time I try to open it shows me this:
<img width="1159" alt="Screen Shot 2023-02-16 at 4 20 30 PM" src="https://user-images.githubusercontent.com/122414464/219489691-76d33b00-092d-4fc0-83e0-fe1b0b094c8a.png">

**Reply by Booplicate** [reply]
I mean in the installer, not in the game.

**Reply by N-1C0lE** [reply]
oh I did the manual install because the installer wasn't working

**Reply by Booplicate** [reply]
I recommended the installer because you're doing something wrong when installing manually. What's your issue with the installer?

**Reply by N-1C0lE** [reply]
well, every time I open the installer it shows me this:

<img width="677" alt="Screen Shot 2023-02-16 at 5 33 56 PM" src="https://user-images.githubusercontent.com/122414464/219502278-42f2352f-c9b3-46b9-9289-c65ec46a7bef.png">

Like what does that mean 😃

**Reply by Booplicate** [reply]
You need to launch it: `./mas-installer-mac` (assuming you `cd` in its directory).

**Reply by N-1C0lE** [reply]
so wait, I move it in the game directory?

**Reply by Booplicate** [reply]
Not sure that would make sense on Mac. You will need to select the game directory when you launch it.

**Reply by N-1C0lE** [reply]
how do I launch it through the game directory tho?

**Reply by N-1C0lE** [reply]
sorry if I'm asking too many questions, it's just that I don't see anyone who installed it through Mac on yt

**Reply by Booplicate** [reply]
You can't launch *`through the game directory`*. You just need to launch it, e.g. `./mas-installer-mac`

**Reply by N-1C0lE** [reply]
yeah but how do I do that ;_;

**Reply by Booplicate** [reply]
Do what?

**Reply by N-1C0lE** [reply]
how do I launch it eg ./mas-installer-mac? I'm more of a visual learner tbh

**Reply by Booplicate** [reply]
You run that command in a terminal of your choice.

**Reply by N-1C0lE** [reply]
the thing is idk how to use the terminal app 💀
what do i type in lmao

**Reply by N-1C0lE** [reply]
Sorry for the long time gap btw 💀💀💀

**Reply by Wingdinggaster656** [reply]
The icon of Terminal looks like:

<img width="109" alt="terminal" src="https://user-images.githubusercontent.com/84446131/228412939-ed5a67b0-a7be-45f2-9af8-505354713210.png">

And as far as I know, this APP comes naturally to any macOS device.

After launching the Terminal, use the `cd` command to navigate to the location of your `mas-installer-mac` (you can use the `ls` command to print the content of the current location. If `mas-installer-mac` is in the displayed list, you have found the correct location), Then type `./mas-installer-mac` to run it.

**Reply by N-1C0lE** [reply]
woah i'm so confused I just found out yesterday that there was a terminal app

Would it be possible to tell me the exact commands step by step to navigate to the location?

**Reply by N-1C0lE** [reply]
Never mind, now it keeps saying Zsh permission denied

**Reply by Wingdinggaster656** [reply]
> Zsh permission denied

A common solution is to execute `chmod +x ./mas-installer-mac`. This will give you a permission so you can run the program without being denied.

**Reply by N-1C0lE** [reply]
Wait I think it worked, it's installing into a non-ddlc directory

**Reply by N-1C0lE** [reply]
now i have a "contents" folder in the downloads section, where do i move it?

**Reply by Wingdinggaster656** [reply]
Why install into a non-ddlc directory? You are supposed to choose a ddlc directory. Do you have DDLC on your computer?

**Reply by N-1C0lE** [reply]
WAIT IT WORKED TYSM
